modules/oe_webtools_analytics/modules/oe_webtools_analytics_rules/src/RuleMatcher.php

```diff
@@ -0,0 +1,273 @@
+  public function getMatchingSection(string $path = NULL): ?string {
+    $current_path = $this->getCurrentPath();
+    $cache = $this->cache->get($current_path) ?: new \stdClass();
+
+    // Check if the cache data for the section has been set, taking into account
+    // that its value might be NULL.
+    if (empty($cache->data) || !array_key_exists('section', $cache->data)) {
+      $rule = $this->getMatchingRule($current_path);
+
+      // Reload the cache object, it is updated when the rule was matched.
+      $cache = $this->cache->get($current_path) ?: new \stdClass();
+
+      $section = $rule instanceof WebtoolsAnalyticsRuleInterface ? $rule->getSection() : NULL;
+      $cache->data = ['section' => $section] + ($cache->data ?? []);
+
+      // We return results based on rule entities. This means that if a rule is
+      // added or deleted, or if any of the existing rules change, the cached
+      // results should be invalidated.
+      $cache->tags = Cache::mergeTags($cache->tags ?? [], $this->getWebtoolsAnalyticsRuleListCacheTags());
+
+      $this->cache->set($current_path, $cache->data, Cache::PERMANENT, $cache->tags);
+    }
+
+    return $cache->data['section'];
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+   */
+  public function getMatchingRule(string $path = NULL): ?WebtoolsAnalyticsRuleInterface {
+    // Default to the current path.
+    if (!$path) {
+      $path = $this->getCurrentPath();
+    }
+
+    $cache = $this->cache->get($path) ?: new \stdClass();
+    if (isset($cache->data['rule'])) {
+      return $this->loadWebtoolsAnalyticsRule($cache->data['rule']);
+    }
+
+    $site_configuration = $this->config->get('system.site');
+    $default_language = $site_configuration->get('default_langcode');
+    $default_language_alias_path = $this->aliasManager->getAliasByPath($this->currentPath->getPath(), $default_language);
+
+    $matching_rule = NULL;
+
+    foreach ($this->loadWebtoolsAnalyticsRules() as $rule) {
+      if (preg_match($rule->getRegex(), $rule->matchOnSiteDefaultLanguage() ? $default_language_alias_path : $path) === 1) {
+        $matching_rule = $rule;
+        break;
+      }
+    }
+
+    $rule_id = $matching_rule instanceof WebtoolsAnalyticsRuleInterface ? $matching_rule->id() : NULL;
+    $cache->data = ['rule' => $rule_id] + ($cache->data ?? []);
+
+    // We return results based on rule entities. This means that if a rule is
+    // added or deleted, or if any of the existing rules change, the cached
+    // results should be invalidated.
+    $cache->tags = Cache::mergeTags($cache->tags ?? [], $this->getWebtoolsAnalyticsRuleListCacheTags());
+
+    // Add the cache tags of the default site configuration if the rule depends
+    // on the default language of the site.
+    $default_language_cache_tags = $matching_rule instanceof WebtoolsAnalyticsRuleInterface && $matching_rule->matchOnSiteDefaultLanguage() ? $site_configuration->getCacheTags() : [];
```

Comment:
Let's use a more clear-to-understand `if()` in here for code readability.

Reply:
It hath been understandified.